src/Selenium2Driver.php

```diff
@@ -246,6 +249,68 @@ protected static function charToOptions($char, $modifier = null)
         return json_encode($options);
     }
 
+    /**
+     * Serialize execute arguments (containing web elements)
+     *
+     * @see https://w3c.github.io/webdriver/#executing-script
+     *
+     * @param array $args
+     *
+     * @return array
+     */
+    private function serializeExecuteArguments(array $args)
+    {
+        foreach ($args as $k => $v) {
+            if ($v instanceof NodeElement) {
+                if (strpos($v->getXpath(), self::XPATH_WEBDRIVER_ELEMENT_PREFIX) === 0) {
+                    [$elementUrlEncoded, $elementIdEncoded] = explode(
+                        '--',
+                        substr($v->getXpath(), strlen(self::XPATH_WEBDRIVER_ELEMENT_PREFIX)),
+                        2
+                    );
+
+                    $args[$k] = new Element(base64_decode($elementUrlEncoded), base64_decode($elementIdEncoded));
+                } else {
+                    $args[$k] = $this->findElement($v->getXpath());
+                }
+            } elseif (is_array($v)) {
+                $args[$k] = $this->serializeExecuteArguments($v);
+            }
+        }
+
+        return $args;
+    }
+
+    /**
+     * Unserialize execute result (containing web elements)
+     *
+     * @param mixed $data
+     *
+     * @return mixed
+     */
+    private function unserializeExecuteResult($data)
+    {
+        if ($data instanceof Element) {
+            $minkSession = \Closure::bind(function () {
+                return $this->session;
+            }, $this, CoreDriver::class)();
+
+            return new NodeElement(
+                self::XPATH_WEBDRIVER_ELEMENT_PREFIX
```

Comment:
would this generated XPATH work when passing this element to a different method of the driver ? Or is it something that you totally made up, and that would only work for passing it to other calls to `executeScript` ?

Reply:
good point - I need to add support for other WebDriver methods, the artificial xpath is supposed to work with any method that does accept `NodeElement` (or 1:1 xpath from it)

Reply:
to me, adding hacks in all method to support a fake XPath is not the solution. Mink does some XPath manipulation when using `NodeElement::find()` for instance.
The XPath passed in NodeElement must be a valid XPath, not a hacked string. Otherwise, you will break Mink.

Reply:
I completely agree, fixed, now xpath is build and NodeElement is created/usable like any other.